### DISCUSSION TITLE
Bun build dts output

### DISCUSSION THREAD

**Discussion by tigawanna** [oven-sh/bun#3281] [Q&A] [answered]
Does bun build have a dts output option?

**Comment by MerlinMason** [comment]
Do you mean to output the type definitions defined in your TypeScript source code? If so, I can't find any options in the `Bun.build` docs related to this which seems surprising, I sort of assumed it would just do this by default.

**Comment by tigawanna** [comment]
Yup that , most build tools don't do it though, it's the slowest part of builds and is usually done with  tsc , am guessing like other bundlers it'll have to be solved with a plugin that'll run after build and run tsc while pointing it to the right output directories, looked into the bun plugin system but didn't get far min try again later

**Comment by wobsoriano** [comment, marked as answer]
You might want to give https://github.com/wobsoriano/bun-plugin-dts a try

**Comment by arshad-yaseen** [comment]
You might use [Bunup](https://github.com/arshad-yaseen/bunup) which simplifies the bun bundler to be used in your project. And also first class support for DTS. Powered by the Bun bundler with Oxc (isolatedDeclaration), you just need the `--dts` flag for DTS generation.

**Reply by tigawanna** [reply]
Thanks for this it looks perfect even similar config to tsup
